### PR TITLE
[JAX] SwiGLU Implementation

### PR DESCRIPTION
**Pull request by phu0ngng**
This PR adds SwiGLU implementation for JAX.
In addition, all previous activation functions were templated to shorten the codebase. 
Tests in `test_custom_call_compute.py` were adapted to be more generalized and address newly added activation.
This PR should be merged after #765.

### CODE REVIEW

File: transformer_engine/jax/csrc/extensions.cpp [hunk 1]
```diff
@@ -66,6 +76,7 @@ PYBIND11_MODULE(transformer_engine_jax, m) {
     m.def("get_device_compute_capability", &GetDeviceComputeCapability);
     m.def("get_cublasLt_version", &cublasLtGetVersion);
     m.def("get_dgelu_dbias_ct_workspace_sizes", &GetDGeluDBiasCastTransposeWorkspaceSizes);
+    m.def("get_dsilu_dbias_ct_workspace_sizes", &GetDSiluDBiasCastTransposeWorkspaceSizes);
```
Comment (denera) [inline transformer_engine/jax/csrc/extensions.cpp:79]: Do all the gated activations have the same workspace size requirement? If so, we could just have a unified `get_gated_activation_workspace_size()` for this.

File: transformer_engine/jax/mlp.py [hunk 2]
```diff
@@ -3,389 +3,143 @@
...
+                "bwd": dsilu },
+    ('silu', 'linear'): { 'fwd': gated_silu,
+                         'bwd': dgated_silu }
+}
+
+activation_fp8_dict = {
+    ('gelu',): { 'fwd': gelu_fp8,
+                'bwd': dgelu_dbias_cast_transpose },
+    ('gelu', 'linear'): { 'fwd': gated_gelu_fp8,
+                          'bwd': dgated_gelu_cast_transpose },
```
Comment (denera) [inline transformer_engine/jax/mlp.py:45]: @phu0ngng and I already talked about this offline as a future PR but just dropping a note here for others.

We should ideally have `fused_layernorm_fp8_mlp()` support biases for every activation, and the activations that do not have `d<act>_dbias_cast_transpose()` implemented should do `d<act>()` + `cast_tranpose_dbias()`.

Biases for these activations are supported in `te.flax.LayerNormMLP` through unfused composition, but we do have `cast_transpose_dbias()` available to us in TE and we should make use of it here to match the TE/PyTorch implementation.

**Review by denera — CHANGES_REQUESTED** [review]

File: transformer_engine/common/include/transformer_engine/activation.h [hunk 3]
```diff
@@ -61,6 +61,50 @@ void nvte_dgeglu(const NVTETensor grad,
                  NVTETensor output,
                  cudaStream_t stream);
 
+/*! \brief Compute SiLU activation of the input.
+ *
+ *  \param[in]     input     Input tensor for GELU activation.
+ *  \param[in,out] output    Output tensor.
+ *  \param[in]     stream    CUDA stream used for the operation.
+ */
+void nvte_silu(const NVTETensor input,
```
[the remark this answers is not in the document]
Comment (phu0ngng) [reply transformer_engine/common/include/transformer_engine/activation.h:70]: JAX uses both `silu` and `swish`.
The reason I chose `silu` over `swish` as it shares the same pattern with other activation functions i.e. `relu`, `gelu`, thus easier for navigation and later development.

[the remark this answers is not in the document]
Comment (denera) [reply transformer_engine/common/include/transformer_engine/activation.h:70]: We should definitely use `silu` in TE/JAX custom ops and modules to match native JAX API (they support `swish` but only as an alias to `silu`).

Using `silu` in TE/common breaks internal consistency though. Our existing device kernel for this is `swish()`, and the existing host function to launch the gated kernel is `nvte_swiglu()`. The new host function to launch the non-gated kernel should be `nvte_swish()` to match the pattern.

File: transformer_engine/jax/mlp.py (hunk 2 above)
Comment (phu0ngng) [reply transformer_engine/jax/mlp.py:45]: Hi, this was addressed in #765. So I marked it as resolved here.

**Comment by denera** [comment]
@phu0ngng I just merged #765. Could you resolve the conflicts with this PR and re-run the CI? We can merge when it comes back clean. Thanks!

**Review by denera — APPROVED** [review]
LGTM!

**Pull request merged**